packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.service.ts

```diff
@@ -32,11 +36,32 @@ export class AdminPanelService {
     private readonly loginTokenService: LoginTokenService,
     private readonly twentyConfigService: TwentyConfigService,
     private readonly domainManagerService: DomainManagerService,
+    private readonly auditService: AuditService,
+    private readonly logger: LoggerService,
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
   ) {}
 
-  async impersonate(userId: string, workspaceId: string) {
+  async impersonate(
+    userId: string,
+    workspaceId: string,
+    impersonatorUserId: string,
+  ) {
+    try {
```

Comment:
This doesn't seem like the standard pattern for input validation in the codebase, usually it's validated at the graphql layer (ImpersonateInput). For impersonatorUserId why not but I don't think it should ever be corrupted unless it's a malicious attempt